numpy/linalg/lapack_lite/clapack_scrub.py

```diff
@@ -221,6 +227,37 @@ def OutOfHeader(line):
         state = state(line)
     return lines.getValue()
 
+def removeSubroutinePrototypes(source):
+    expression = re.compile(
+        '/\* Subroutine \*/^\s*(?:(?:inline|static)\s+){0,2}(?!else|typedef|return)\w+\s+\*?\s*(\w+)\s*\([^0]+\)\s*;?'
+    )
+    lines = LineQueue()
+    for line in BytesIO(source):
+        if not expression.match(line):
+            lines.add(line)
+
+    return lines.getValue()
+
+def removeBuiltinFunctions(source):
+    lines = LineQueue()
+    def LookingForBuiltinFunctions(line):
+        if line.strip() == '/* Builtin functions */':
+            return InBuiltInFunctions
+        else:
+            lines.add(line)
+            return LookingForBuiltinFunctions
+
+    def InBuiltInFunctions(line):
+        if line.strip() == '':
+            return LookingForBuiltinFunctions
+        else:
+            return InBuiltInFunctions
+
+    state = LookingForBuiltinFunctions
+    for line in BytesIO(source):
+        state = state(line)
```

Comment:
Neat state machine construct.  Inspired from elsewhere in the file, or just the way you normally do this?

Reply:
Essentially copypasted from `removeHeader` above.